tensorflow/python/ops/image_ops.py

```diff
@@ -427,44 +433,57 @@ def pad_to_bounding_box(image, offset_height, offset_width, target_height,
 
   Raises:
     ValueError: If the shape of `image` is incompatible with the `offset_*` or
-      `target_*` arguments
+      `target_*` arguments, or either `offset_height` or `offset_width` is
+      negative.
   """
   image = ops.convert_to_tensor(image, name='image')
-  _Check3DImage(image, require_static=True)
+  _Check3DImage(image, require_static=False)
   height, width, depth = _ImageDimensions(image)
 
-  if target_width < width:
-    raise ValueError('target_width must be >= width')
-  if target_height < height:
-    raise ValueError('target_height must be >= height')
-
   after_padding_width = target_width - offset_width - width
   after_padding_height = target_height - offset_height - height
 
-  if after_padding_width < 0:
+  if not isinstance(offset_height, ops.Tensor) and offset_height < 0:
```

Comment:
Using `isinstance(something, ops.Tensor)` here is too clumsy.

Solution 1:
Use some helper function to check whether the value is a "literal" or "symbol".

Solution 2:
Make the framework automatically evaluate all ops that have constant inputs at graph construction time (like C++ constexpr).